src/MovieBox.jsx

```diff
@@ -0,0 +1,271 @@
+                const searchURL = buildMovieSearchURL()
+                let response = await fetch(searchURL, options)
+                if (!response.ok) {throw new Error('Failed to fetch movies by search')}
+                const data = await response.json()
+                return data
+            } catch (error) {
+                console.error(error)
+            }
+        }
+        const processMoviesBySearch = (moviesData) => {
+            if (moviesData.total_pages === page) {setMorePages(false)}
+            let newMovies = {}
+            let newOrder = []
+            // TMDB seems to be giving 1-2 repeats from the end of the last page on load more
+            moviesData.results.map( (movie) => {
+                if (!order.includes(movie.id) && !newOrder.includes(movie.id)) {
+                    newMovies[movie.id] = movie
+                    newOrder.push(movie.id)
+                }
+            })
+            setOriginalOrder([...originalOrder, ...newOrder])
+            newMovies = {...movies, ...newMovies}
+            newOrder = [...order, ...newOrder]
+            setOrder(newOrder)
+            setMovies(newMovies)
+        }
+
+        let moviesData = await fetchMoviesBySearch()
+        processMoviesBySearch(moviesData)
+    }
+
+    // Sorts movies based on sort mode (either title, release date, or vote average
+    // Works on the movies state dict and the order state list
+    const sortMovieOrder = () => {
+        if (movies.length === 0) {
+            return;
+        }
+        // Split movies dict into keys (id) and values (movie)
+        let movieEntries = Object.entries(movies)
+        // Listened to feedback about using switch/case for repeated if/else if or where it's easier to read
+        switch (sortMode) {
+            case "title":
+                movieEntries.sort((left_entry, right_entry) => {
+                    const left = left_entry[1]
+                    const right = right_entry[1]
+                    if (left.title < right.title) {return -1;}
+                    if (left.title > right.title) {return 1;}
+                    return 0;
+                })
+                break;
+            case "release":
+                // Listened to feedback about shortening sorting comparator
+                // Note: can't subtract strings in JS
+                movieEntries.sort((left_entry, right_entry) => {
+                    const left = left_entry[1]
+                    const right = right_entry[1]
+                    left.date = new Date(left.release_date)
+                    right.date = new Date(right.release_date)
+                    return right.date - left.date
+                })
+                break;
+            case "vote":
+                movieEntries.sort((left_entry, right_entry) => {
+                    const left = left_entry[1]
+                    const right = right_entry[1]
+                    return right.vote_average-left.vote_average
+                })
```

Comment:
Since the left/right consts are only used once after and the left_entry and right_entry are well named, for conciseness this could also be condensed to:
```suggestion
                movieEntries.sort((left_entry, right_entry) => {
                    return right_entry[1].vote_average-left_entry[1].vote_average
                })
```